docs/develop/natmod.rst

```diff
@@ -0,0 +1,191 @@
+Native machine code in .mpy files
+=================================
+
+This section describes how to build and work with .mpy files that contain native
+machine code from a language other than Python.  This allows you to
+write code in a language like C, compile and link it into a .mpy file, and then
+import this file like a normal Python module.  This can be used for implementing
+functionality which is performance critical, or for including an existing
+library written in another language.  One of the main advantages of this is that
+native machine code can be imported by a script dynamically, without the need
+to rebuild the main MicroPython firmware.
```

Comment:
Maybe a paragraph to compare to `develop/cmodules.rst`

"This is in contrast to :ref:`cmodules` which also allows you to write modules in C, but these must be built into the main firmware image."

Perhaps also an equivalent paragraph could be added to `cmodules.rst` pointing back the other way.